pkg/pillar/hypervisor/xen.go

```diff
@@ -581,22 +581,25 @@ func (ctx xenContext) PCIRelease(long string) error {
 }
 
 func (ctx xenContext) IsDeviceModelAlive(domid int) bool {
-	// create pgrep command to see if dataplane is running
-	match := fmt.Sprintf("domid %d", domid)
-	cmd := wrap.Command("pgrep", "-f", match)
+	if isDomainStillRunning(domid) { //if domain is up but not healthy, then do not verify qemu as domain is shutting down.
+		// create pgrep command to see if dataplane is running
+		match := fmt.Sprintf("domid %d", domid)
+		cmd := wrap.Command("pgrep", "-f", match)
 
-	// pgrep returns 0 when there is atleast one matching program running
-	// cmd.Output returns nil when pgrep returns 0, otherwise pids.
-	out, err := cmd.Output()
+		// pgrep returns 0 when there is atleast one matching program running
+		// cmd.Output returns nil when pgrep returns 0, otherwise pids.
+		out, err := cmd.Output()
 
-	if err != nil {
-		log.Infof("IsDeviceModelAlive: %s process is not running: %s",
-			match, err)
-		return false
+		if err != nil {
+			log.Infof("IsDeviceModelAlive: %s process is not running: %s",
+				match, err)
+			return false
+		}
+		log.Infof("IsDeviceModelAlive: Instances of %s is running: %s",
+			match, out)
+		return true
 	}
-	log.Infof("IsDeviceModelAlive: Instances of %s is running: %s",
-		match, out)
-	return true
+	return false
```

Comment:
Close, but still still results in the error verifyStatus(%s) qemu crashed.
You need to check isDomainStillRunning before even bothering to check for qemu in the caller.

Reply:
Changed.